pkg/request/pool.go

```diff
@@ -67,114 +63,117 @@ type requestItem struct {
 
 // PoolOptions is the pool configuration
 type PoolOptions struct {
-	MaxSize               int
-	BatchMaxSize          int
+	MaxSize               uint64
+	BatchMaxSize          uint64
+	BatchMaxSizeBytes     uint64
+	RequestMaxBytes       uint64
 	SubmitTimeout         time.Duration
-	AutoRemoveTimeout     time.Duration
-	OnFirstStrikeTimeout  func([]byte)
+	BatchTimeout          time.Duration
+	OnFirstStrikeTimeout  func([]byte, types.RequestInfo)
```

Comment:
please let's define types.RequestInfo locally in this package and not reference the `api` or `types` ones.